common/libs/contract.js

```diff
@@ -65,25 +65,16 @@ export default class Contract {
       throw new Error('Invalid number of arguments');
     }
 
-    return method.inputs
-      .map((input, idx) => this.encodeArg(input, args[idx]))
-      .join('');
+    const inputTypes = method.inputs.map(input => input.type);
+    return abi.rawEncode(inputTypes, args).toString('hex');
   }
 
-  encodeArg(input: ABITypedSlot, arg: any): string {
-    switch (input.type) {
-      case 'address':
-      case 'uint160':
-        assertString(arg);
-        return setLengthLeft(toBuffer(arg), 32).toString('hex');
-      case 'uint256':
-        if (arg instanceof Big) {
-          arg = '0x' + arg.toString(16);
-        }
-        assertString(arg);
-        return setLengthLeft(toBuffer(arg), 32).toString('hex');
-      default:
-        throw new Error(`Dont know how to handle abi type ${input.type}`);
-    }
+  decodeArgs(method: ABIMethod, argData: string) {
+    // Remove method selector from data, if present
+    argData = argData.replace(`0x${this.getMethodSelector(method)}`, '');
+    // Convert argdata to a hex buffer for ethereumjs-abi
+    const argBuffer = new Buffer(argData, 'hex');
+    // Decode!
+    return abi.rawDecode(this.getMethodTypes(method), argBuffer);
```

Comment:
```
 FAIL  spec/libs/erc20.spec.js
  ● ERC20 › decodeTransfer › should return the correct transaction given some data

    expect(received).toBe(expected)

    Expected value to be (using ===):
      "0x7cB57B5A97eAbe94205C07890BE4c1aD31E486A8"
    Received:
      "0x0"
```

When I comment the line out. Going to add some more tests just to make sure this isn't something wrong with the ERC20 implementation, but I'm pretty sure it's from the method selector screwing it up.